### PR TITLE
Added dynamic authentication carbon feature and updated pom

### PR DESCRIPTION
**Pull request by milanperera**

### CODE REVIEW

File: modules/apps/jax-rs/mdm-admin/src/main/java/org/wso2/carbon/mdm/api/MobileDevice.java [hunk 1]
```diff
@@ -58,7 +59,9 @@
                 allDevices = service.getDevicesOfUser(user);
             } else if (role != null){
                 allDevices = service.getAllDevicesOfRole(role);
-            } else {
+            } else if (status != null) {
+				allDevices = service.getDevicesByStatus(status);
+			} else {
```
Comment (DilanUA) [inline modules/apps/jax-rs/mdm-admin/src/main/java/org/wso2/carbon/mdm/api/MobileDevice.java]: unnecessary indentation

File: modules/mobile-agents/android/client/client/src/main/java/org/wso2/emm/agent/services/Operation.java [hunk 2]
```diff
@@ -1016,6 +1019,13 @@ private void installApplication(JSONObject data, org.wso2.emm.agent.beans.Operat
 		}
 	}
 
+	public void disenrollDevice(org.wso2.emm.agent.beans.Operation operation) {
```
Comment (DilanUA) [inline modules/mobile-agents/android/client/client/src/main/java/org/wso2/emm/agent/services/Operation.java]: better if this method can return disenrollment status also.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java [hunk 3]
```diff
@@ -822,6 +822,49 @@ public Response setWebClip(@HeaderParam("Accept") String acceptHeader,
         }
     }
 
+	@POST
+	@Produces(MediaType.APPLICATION_JSON)
+	@Path("disenroll")
+	public Response setDisenrollment(@HeaderParam("Accept") String acceptHeader,
```
Comment (DilanUA) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java]: While we allow any type of response via the accept header,  why do we also limit response to JSON?

File: modules/apps/jax-rs/mdm-admin/src/main/java/org/wso2/carbon/mdm/api/MobileDevice.java (hunk 1 above)
Comment (milanperera) [inline modules/apps/jax-rs/mdm-admin/src/main/java/org/wso2/carbon/mdm/api/MobileDevice.java]: Fixed in [#70]

File: modules/mobile-agents/android/client/client/src/main/java/org/wso2/emm/agent/services/Operation.java (hunk 2 above)
Comment (milanperera) [inline modules/mobile-agents/android/client/client/src/main/java/org/wso2/emm/agent/services/Operation.java]: This method is called remotely and status will send back in the response itself.

File: modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java (hunk 3 above)
Comment (milanperera) [inline modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/OperationMgtService.java]: Fixed in [#70]

**Pull request merged**